detect-nuget-inspector/detect-nuget-inspector/Inspection/Inspectors/ProjectInspector.cs

```diff
@@ -70,6 +72,69 @@ public ProjectInspector(ProjectInspectionOptions options, NugetSearchService nug
             {
                 Options.VersionName = InspectorUtil.GetProjectAssemblyVersion(Options.ProjectDirectory);
             }
+            
+            if (String.IsNullOrWhiteSpace(Options.DirectoryPackagesPropsPath))
+            {
+                Options.DirectoryPackagesPropsPath = CreateDirectoryPackagesPropsPath(Options.ProjectDirectory);
+            }
+        }
+        
+        public ProjectInspector(ProjectInspectionOptions options, NugetSearchService nugetService, HashSet<PackageId> packages, bool checkVersionOverride)
```

Comment:
This is also adding one more constructor to extend support for CPM and passing the packages along the ProjectInspector.